diff --git a/node_modules/react-native-ble-plx/android/src/main/java/com/bleplx/utils/SafePromise.java b/node_modules/react-native-ble-plx/android/src/main/java/com/bleplx/utils/SafePromise.java
index 0203473..c69c7d0 100644
--- a/node_modules/react-native-ble-plx/android/src/main/java/com/bleplx/utils/SafePromise.java
+++ b/node_modules/react-native-ble-plx/android/src/main/java/com/bleplx/utils/SafePromise.java
@@ -20,21 +20,21 @@ public class SafePromise {
     }
   }
 
-  public void reject(String code, String message) {
+  public void reject(@Nullable String code, String message) {
     if (isFinished.compareAndSet(false, true)) {
-      promise.reject(code, message);
+      promise.reject(normalizeCode(code), message);
     }
   }
 
-  public void reject(String code, Throwable e) {
+  public void reject(@Nullable String code, Throwable e) {
     if (isFinished.compareAndSet(false, true)) {
-      promise.reject(code, e);
+      promise.reject(normalizeCode(code), e);
     }
   }
 
-  public void reject(String code, String message, Throwable e) {
+  public void reject(@Nullable String code, String message, Throwable e) {
     if (isFinished.compareAndSet(false, true)) {
-      promise.reject(code, message, e);
+      promise.reject(normalizeCode(code), message, e);
     }
   }
 
@@ -50,4 +50,8 @@ public class SafePromise {
       promise.reject(reason);
     }
   }
+
+  private String normalizeCode(@Nullable String code) {
+    return code != null ? code : "BleError";
+  }
 }
